wmfdata/hive.py

```diff
@@ -1,40 +1,115 @@
+    # this will result in a error when Pandas tries to read the resulting concatenated output (unless the queries
+    # happen to produce the same number of columns).
+    #
+    # Ideally, we would return only the last query's results or throw a clearer error ourselves. However,
+    # there's no simple way to determine if multiple queries have been passed or separate their output,
+    # so it's not worth the effort.
+    merged_commands = ";\n".join(commands)
     
-    spark.start_session_timeout(spark_session)
+    try:
+        # Create temporary files in current working directory to write to:
+        cwd = os.getcwd()
+        query_fd, query_path = tempfile.mkstemp(suffix=".hql", dir=cwd)
+        results_fd, results_path = tempfile.mkstemp(suffix=".tsv", dir=cwd)
+
+        # Write the Hive query:
+        with os.fdopen(query_fd, 'w') as fp:
+            fp.write(merged_commands)
+
+        # Execute the Hive query:
+        shell_command = shell_command.format(heap_size, query_path)
+        hive_call = subprocess.run(shell_command, shell=True, stdout=results_fd, stderr=subprocess.PIPE)
+        if hive_call.returncode == 0:
+            # Read the results upon successful execution of cmd:
+            if format == "pandas":
+                try:
+                    result = pd.read_csv(results_path, sep='\t')
+                except pd.errors.EmptyDataError:
+                    # The command had no output
+                    pass
+            else:
+                # If user requested "raw" results, read the text file as-is:
+                with open(results_path, 'r') as file:
+                    content = file.read()
+                    # If the statement had output:
+                    if content:
+                        result = content
+        # If the hive call has not completed successfully
+        else:
+            # Remove logspam from the standard error so it's easier to see the actual error
+            stderr = iter(hive_call.stderr.decode().splitlines())
+            cleaned_stderr = ""
+            for line in stderr:
+                filter = r"JAVA_TOOL_OPTIONS|parquet\.hadoop|WARN:|:WARN|SLF4J"
+                if re.search(filter, line) is None:
+                    cleaned_stderr += line + "\n"
+
+            raise ChildProcessError(
+                "The Hive command line client encountered the following error:\n{}".format(cleaned_stderr)
+            )
+    finally:
+        # Remove temporary files:
+        os.unlink(query_path)
+        os.unlink(results_path)
+
     return result
 
+def run(commands, format="pandas", engine="cli"):
+    """
+    Runs SQL commands against the Hive tables in the Data Lake. Currently, this simply passes the commands to the `run_cli` function.
+    """
+
+    if format not in ["pandas", "raw"]:
+        raise ValueError("The `format` should be either `pandas` or `raw`.")
+    if engine not in ("cli"):
```

Comment:
should be `["cli"]`

Reply:
Ahh, it's not a tuple! I went ahead and changed all these checks to use lists